ibis/backends/sql/rewrites.py

```diff
@@ -249,17 +249,17 @@ def merge_select_select(_, **kwargs):
     return result if complexity(result) <= complexity(_) else _
 
 
-def extract_ctes(node):
-    result = []
+def extract_ctes(node: ops.Relation) -> Set[ops.Relation]:
```

Comment:
This returns an actual instance of `set`, `collections.abc.Set` is an ABC. You can just use `set[ops.Relation]` here instead, no need for the ABC semantics.

```suggestion
def extract_ctes(node: ops.Relation) -> set[ops.Relation]:
```